src/wmtk/simplex/SimplexCollection.hpp

```diff
@@ -0,0 +1,75 @@
+#pragma once
+
+#include <vector>
+#include <wmtk/Mesh.hpp>
+#include "Simplex.hpp"
+#include "internal/SimplexEqualFunctor.hpp"
+#include "internal/SimplexLessFunctor.hpp"
+
+namespace wmtk::simplex {
+class SimplexCollection
+{
+public:
+    SimplexCollection(const Mesh& mesh)
+        : m_mesh{mesh}
+        , m_simplex_is_less(mesh)
```

Comment:
TBH this member just stores a single ref - might as well make a private member that creates the function rather than store the member (it's like storing 3 pointers to the same mesh?)